brigade.js

```diff
@@ -72,6 +72,32 @@ const lintJSJob = (e, p) => {
 }
 jobs[lintJSJobName] = lintJSJob;
 
+const testIntegrationJobName = "test-integration";
+const testIntegrationJob = (e, p) => {
+  let kind = new KindJob(testIntegrationJobName);
+  kind.mountPath = localPath;
+  kind.tasks.push(
+    // Install git and golang deps
+    "apk add --update --no-cache git gcc musl-dev",
+    // Install helm
+    `curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 \
+      && chmod 700 get_helm.sh \
+      && ./get_helm.sh`,
+    `cd ${localPath}`,
+    // It would be more efficient to mount built images from associated
+    // build jobs; however, they currently use kaniko, which doesn't preserve
+    // images in the local Docker cache.
+    "make hack-build-images hack-load-images",
```

Comment:
This entire comment is non-blocking...

A `KindJob` must be using DooD to do this?

DooD weirds me out because it clutters the image cache in the k8s node we're running on.

I wonder if we could use a kind node that has a registry built in, and then with `DOCKER_REGISTRY` and `IMAGE_PULL_POLICY ` set correctly, `make push-images hack-deploy test-integration` would do the trick?

The benefit would be that when the kind node node goes away, so to do all the images that we don't care about keeping.

Again... non-blocking.

Reply:
Good concern.  The KindJob actually sets up DinD: https://github.com/brigadecore/brigade-utils/blob/master/src/kind.ts#L73  So, at least for these Brigade 2 images, no image cache-crowding on the k8s node will occur.  (For running kind itself in the KindJob, there are some host mounts necessary).

It could still be interesting to see if launching an internal registry in Kind helps efficiency, etc.

Reply:
Sorry... I had a mental lapse. For some reason I couldn't see how this could be working with DinD, but I see it now.

DinD doesn't both me as much.

Forget I asked about this.